In [2]:
import joblib
import numpy as np
import pandas as pd

### Load file asap 1-8, except asap 6

In [3]:
asap1 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap1')
asap2 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap2')
asap3 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap3')
asap4 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap4')
asap5 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap5')
asap7 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap7')
asap8 = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap8')

asap1 = pd.DataFrame(asap1)
asap2 = pd.DataFrame(asap2)
asap3 = pd.DataFrame(asap3)
asap4 = pd.DataFrame(asap4)
asap5 = pd.DataFrame(asap5)
asap7 = pd.DataFrame(asap7)
asap8 = pd.DataFrame(asap8)

### sample 50 essays from each file (to have total 350 off topic essays)

In [4]:
asap1_sample = asap1.sample(n=50, random_state=42).to_numpy()
asap2_sample = asap2.sample(n=50, random_state=42).to_numpy()
asap3_sample = asap3.sample(n=50, random_state=42).to_numpy()
asap4_sample = asap4.sample(n=50, random_state=42).to_numpy()
asap5_sample = asap5.sample(n=50, random_state=42).to_numpy()
asap7_sample = asap7.sample(n=50, random_state=42).to_numpy()
asap8_sample = asap8.sample(n=50, random_state=42).to_numpy()

### Concatenate off-topic essays

In [5]:
off_topic = np.concatenate((asap1_sample, asap2_sample, asap3_sample, asap4_sample, asap5_sample, asap7_sample, asap8_sample), axis=0)
off_topic.shape

(350, 780)

In [97]:
joblib.dump(off_topic, 'essay_350_offtopic_780')

['essay_350_offtopic_780']

### load file

In [6]:
off_topic = joblib.load('essay_350_offtopic_780')

### Create asap6 780-dim feature, including the off topic (in total 2150 essays)

In [7]:
x = joblib.load('essay_ease10_sbert768_simbow_langerr_780_normalized_asap6')
y = joblib.load("score_asap6")

In [8]:
feats = np.concatenate((x, off_topic), axis=0)
feats.shape

(2150, 780)

In [9]:
joblib.dump(feats, 'essay_asap6_780_with350offtopic')

['essay_asap6_780_with350offtopic']

### Create asap6 score, with 0 (zero) scores for 350 topic (total 2150 esay scores)

In [10]:
zeros = np.zeros(350)
print(zeros.shape)
y = np.concatenate((y, zeros), axis=0)
print(y.shape)

(350,)
(2150,)


In [11]:
joblib.dump(y, 'score_asap6_with350offtopic')

['score_asap6_with350offtopic']